### PR TITLE
feat(joml): align Region related classes

### PR DESCRIPTION
**Pull request by jdrueckert**
- needs to be merged with https://github.com/Terasology/Scenario/pull/48

### CODE REVIEW

File: src/main/java/org/terasology/structureTemplates/util/ProtectedRegionUtility.java [hunk 1]
```diff
@@ -17,10 +17,12 @@
 
 import org.terasology.entitySystem.entity.EntityRef;
 import org.terasology.logic.location.LocationComponent;
+import org.terasology.math.JomlUtil;
 import org.terasology.math.Region3i;
 import org.terasology.math.geom.Quat4f;
 import org.terasology.math.geom.Vector3i;
 import org.terasology.structureTemplates.components.ProtectedRegionsComponent;
+import org.terasology.world.block.BlockRegion;
```
Comment (pollend) [inline src/main/java/org/terasology/structureTemplates/util/ProtectedRegionUtility.java]:
```suggestion
import org.terasology.entitySystem.entity.EntityRef;
import org.terasology.math.JomlUtil;
import org.terasology.math.geom.Vector3i;
import org.terasology.structureTemplates.components.ProtectedRegionsComponent;
import org.terasology.world.block.BlockRegion;
```

**Review by pollend — CHANGES_REQUESTED** [review]

**Comment by jdrueckert** [comment]
@pollend can be re-reviewed

**Comment by jdrueckert** [comment]
@pollend this still needs a review, so that we can get https://github.com/Terasology/Scenario/pull/48 in

File: src/main/java/org/terasology/structureTemplates/util/ProtectedRegionUtility.java [hunk 2]
```diff
@@ -35,12 +34,13 @@
...
-        List<Region3i> protectedRegions = protectedRegionsComponent.regions;
+        ProtectedRegionsComponent protectedRegionsComponent =
+                regionEntity.getComponent(ProtectedRegionsComponent.class);
+        List<BlockRegion> protectedRegions = protectedRegionsComponent.regions;
         if (protectedRegions != null) {
-            for (Region3i region : protectedRegions) {
+            for (BlockRegion region : protectedRegions) {
                 for (Vector3i position : positions) {
-                    if (region.encompasses(position)) {
+                    if (region.containsPoint(JomlUtil.from(position))) {
```
Comment (pollend) [inline src/main/java/org/terasology/structureTemplates/util/ProtectedRegionUtility.java]: BlockRegion also has a containsBlock function that is non inclusive. double check with encompasses to make sure the implementation is correct.

**Comment by jdrueckert** [comment]
Changes already done

**Pull request closed without merging**